### PR TITLE
Fix tzdata cross-compilation

### PR DESCRIPTION
**Pull request by anmonteiro**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/pull/113295 tree-wide change broke tzdata's installation when cross-compiling as the target `zic` needs to be run on the build platform.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by r-burns — APPROVED** [review]
Fixes pkgsCross.{aarch64-multiplatform,powernv,riscv64}.tzdata. Thanks!

**Comment by SuperSandro2000** [comment]
Can you please add a comment that we do not change this in the future?

**Comment by anmonteiro** [comment]
added a comment.

File: pkgs/data/misc/tzdata/default.nix [hunk 1]
```diff
@@ -42,10 +42,11 @@ stdenv.mkDerivation rec {
 
   installFlags = [ "ZIC=./zic-native" ];
 
+  # NOTE: in cross-compilation scenarios, this is built for the build platform.
```
Comment (SuperSandro2000) [inline pkgs/data/misc/tzdata/default.nix:45]:
```suggestion
```

File: pkgs/data/misc/tzdata/default.nix [hunk 2]
```diff
@@ -42,10 +42,11 @@ stdenv.mkDerivation rec {
...
   preInstall = ''
      mv zic.o zic.o.orig
      mv zic zic.orig
-     make $makeFlags cc=${stdenv.cc.targetPrefix}cc AR=${stdenv.cc.targetPrefix}ar zic
+     make $makeFlags cc=cc AR=ar zic
```
Comment (SuperSandro2000) [inline pkgs/data/misc/tzdata/default.nix:49]:
```suggestion
     # NOTE: in cross-compilation scenarios, this is built for the build platform.
     make $makeFlags cc=cc AR=ar zic
```
just to make it super clear.

**Comment by anmonteiro** [comment]
What's the process for getting this into master?

**Comment by r-burns** [comment]
Every couple weeks or so, a batch of commits from staging are merged into staging-next. That batch runs on hydra, and once any regressions are sorted out, it's merged into master. There's already a batch in staging-next, so this fix will probably be part of the next one.
More info here: https://github.com/NixOS/rfcs/blob/master/rfcs/0026-staging-workflow.md

**Comment by dotlambda** [comment]
Is https://github.com/NixOS/nixpkgs/pull/114052 preferable to this?

**Comment by Mic92** [comment]
> Is #114052 preferable to this?

I think so. It looks more purposefully without the need for an additional comment.

**Comment by dotlambda** [comment]
I reverted this in 26c9976e500c25dfcb5b0a48c307d9549a1df729 and 109cbc400f4dd472d91f1f7768f42d7faac0cddb.

**Pull request merged**